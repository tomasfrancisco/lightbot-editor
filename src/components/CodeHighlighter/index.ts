export * from "./CodeHighlighter";
export * from "./CodeHighlighterButton";
