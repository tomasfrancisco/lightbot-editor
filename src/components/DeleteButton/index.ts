export * from "./DeleteButton";
export * from "./enum/DeleteEnums";
export * from "./constants/constants";
