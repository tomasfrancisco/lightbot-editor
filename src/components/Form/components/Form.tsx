import { Form as AntdForm } from "antd";
import { FormProps as AntdFormProps } from "antd/lib/form";
import * as React from "react";
import { css } from "react-emotion";

const formStyle = css`
  min-width: 314px;
  margin-left: 16px;
  margin-top: 10px;
`;

export type FormProps = AntdFormProps & {};

export const Form = (props: FormProps) => {
  return <AntdForm className={formStyle} {...props} />;
};
