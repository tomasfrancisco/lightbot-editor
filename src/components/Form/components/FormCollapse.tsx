import { Collapse } from "antd";
import styled from "react-emotion";

export const FormCollapse = styled(Collapse)`
  background-color: transparent;
`;
