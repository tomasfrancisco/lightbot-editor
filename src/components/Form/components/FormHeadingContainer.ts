import styled from "@emotion/styled";

export const FormHeadingContainer = styled("div")`
  margin-bottom: 24px;
`;
