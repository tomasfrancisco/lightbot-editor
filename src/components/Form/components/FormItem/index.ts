export * from "./FormItem";
export * from "./FormItemContainer";
export * from "./FormItemDeleteIcon";
export * from "./FormItemIcon";
