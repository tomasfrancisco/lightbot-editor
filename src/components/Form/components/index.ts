export * from "./FormPanel";
export * from "./FormCollapse";
export * from "./Form";
export * from "./FormItem";
export * from "./FormGroup";
export * from "./FormActionLink";
export * from "./FormHeadingContainer";
export * from "./FormLogoWrapper";
