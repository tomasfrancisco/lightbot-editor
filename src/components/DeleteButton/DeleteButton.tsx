import { Button as AntdButton, Dropdown, Icon, Menu, Popconfirm } from "antd";
import * as React from "react";
import styled, { css } from "react-emotion";
import { DeleteActionEnum } from "~/components/DeleteButton";
import theme from "~/config/theme.js";

const WarningIcon = styled(Icon)`
  color: ${theme["error-color"]} !important;
`;

const deleteButtonStyle = css`
  color: ${theme["error-color"]};

  &:hover,
  &:active,
  &:focus {
    color: ${theme["error-color"]};
    border-color: ${theme["error-color"]};
  }
`;

const deleteDropdownButtonStyle = css`
  & > button {
    color: ${theme["error-color"]};

    &:hover,
    &:active,
    &:focus {
      color: ${theme["error-color"]};
      border-color: ${theme["error-color"]};
    }
  }
`;

export type DeleteAction = {
  key: DeleteActionEnum;
  optionTitle?: string;
  confirmationMessage: string;
};

type DeleteButtonProps = {
  dropdownActions?: DeleteAction[];
  mainAction: DeleteAction;
  disabled?: boolean;
  onConfirm(deleteAction: DeleteAction): void;
};

type DeleteButtonState = {
  selectedAction: DeleteAction;
  isConfirmVisible: boolean;
};

export class DeleteButton extends React.Component<
  DeleteButtonProps,
  DeleteButtonState
> {
  constructor(props) {
    super(props);

    const { mainAction } = this.props;

    this.state = {
      isConfirmVisible: false,
      selectedAction: mainAction
    };
  }

  public render() {
    const { isConfirmVisible, selectedAction } = this.state;

    return (
      <Popconfirm
        title={selectedAction.confirmationMessage}
        visible={isConfirmVisible}
        onConfirm={this.onConfirm}
        onCancel={this.onCancel}
        icon={this.renderPopconfirmIcon}
        arrowPointAtCenter={true}
        placement="bottom"
        okType="danger"
        okText="Yes"
        cancelText="Cancel"
      >
        {this.renderCorrectOption()}
      </Popconfirm>
    );
  }

  private renderCorrectOption = () => {
    const { dropdownActions } = this.props;
    return dropdownActions
      ? this.renderWithDropDown()
      : this.renderWithoutDropDown();
  };

  private renderWithoutDropDown = () => {
    const { disabled } = this.props;
    return (
      <AntdButton
        disabled={disabled}
        onClick={this.onDefaultClick}
        className={deleteButtonStyle}
      >
        <Icon type="delete" />
        Delete
      </AntdButton>
    );
  };

  private renderWithDropDown = () => {
    const { disabled } = this.props;

    return (
      <Dropdown.Button
        overlay={this.renderMenu()}
        disabled={disabled}
        onClick={this.onDefaultClick}
        className={deleteDropdownButtonStyle}
      >
        <Icon type="delete" />
        Delete
      </Dropdown.Button>
    );
  };

  private renderMenu = () => {
    return (
      <Menu onClick={this.onMenuItemClick}>{this.renderMenuOptions()}</Menu>
    );
  };

  private renderMenuOptions = () => {
    const { dropdownActions } = this.props;
    return dropdownActions!.map(action => {
      return (
        <Menu.Item key={action.key}>
          <Icon type="arrow-right" />
          {action.optionTitle}
        </Menu.Item>
      );
    });
  };

  private get renderPopconfirmIcon() {
    return <WarningIcon type="warning" />;
  }

  private onMenuItemClick = ({ item }) => {
    const { dropdownActions } = this.props;
    const { index } = item.props;

    this.setState({
      isConfirmVisible: true,
      selectedAction: dropdownActions![index]
    });
  };

  private onDefaultClick = () => {
    const { mainAction } = this.props;
    this.setState({
      isConfirmVisible: true,
      selectedAction: mainAction
    });
  };

  private onConfirm = () => {
    this.setState({
      isConfirmVisible: false
    });

    const { onConfirm } = this.props;
    const { selectedAction } = this.state;
    if (onConfirm) {
      onConfirm(selectedAction);
    }
  };

  private onCancel = () => {
    this.setState({
      isConfirmVisible: false
    });
  };
}
