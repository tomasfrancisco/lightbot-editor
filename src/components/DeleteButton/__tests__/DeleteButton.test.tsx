import { shallow } from "enzyme";
import * as React from "react";
import { DeleteAction, DeleteButton } from "src/components/DeleteButton";

const deleteAction: DeleteAction = {
  confirmationMessage: "Are you sure",
  key: "DEFAULT_DELETE",
};

const deleteConfirmation = (selectedAction: DeleteAction) => {
  const action = selectedAction;
};

describe("components/Button", () => {
  test("renders", () => {
    const tree = shallow(
      <DeleteButton mainAction={deleteAction} disabled={false} onConfirm={deleteConfirmation} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
