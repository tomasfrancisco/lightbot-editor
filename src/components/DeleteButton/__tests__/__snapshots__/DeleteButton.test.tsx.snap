// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button renders 1`] = `
<Popconfirm
  arrowPointAtCenter={true}
  cancelText="Cancel"
  icon={
    <Styled(Icon)
      type="warning"
    />
  }
  okText="Yes"
  okType="danger"
  onCancel={[Function]}
  onConfirm={[Function]}
  placement="bottom"
  title="Are you sure"
  transitionName="zoom-big"
  trigger="click"
  visible={false}
>
  <Button
    block={false}
    className="css-48x2js"
    disabled={false}
    ghost={false}
    loading={false}
    onClick={[Function]}
  >
    <Icon
      type="delete"
    />
    Delete
  </Button>
</Popconfirm>
`;
